yaml/com/ibm/VPD/FRUManager.interface.yaml

```diff
@@ -0,0 +1,122 @@
+description: >
+    Implement to manage VPD in the system.
+
+methods:
+    - name: WriteKeyword
+      description: >
+          Method to update the keyword value with the given value. Updates the
+          keyword on hardware and also updates on the cache, if cache
+          information is present in the config JSON.
+      parameters:
+          - name: path
+            type: string
+            description: >
+                Dbus/EEPROM path of the FRU whose keyword needs to be updated.
+          - name: data
+            type:
+                variant[struct[string,string,array[byte]],
+                struct[string,array[byte]]]
+            description: >
+                Information required to update the value of a keyword and the
+                data to write.
+      returns:
+          - name: bytesWritten
+            type: int64
+            description: >
+                Returns -1 on failure otherwise, returns the number of bytes
+                written.
+      errors:
+          - xyz.openbmc_project.Common.Error.InvalidArgument
+
+    - name: ReadKeyword
+      description: >
+          Method to read VPD keyword value. Keyword can be read from hardware or
+          from Dbus.
+      parameters:
+          - name: path
+            type: string
+            description: >
+                Dbus/EEPROM Path of the FRU whose keyword needs to be read.
+          - name: data
+            type: variant[struct[string,string],struct[string]]
+            description: >
+                Information required to read value of keyword.
+          - name: target
+            type: boolean
+            description: >
+                Default will be false and will read from hardware in that case,
+                if true will read from cache.
+      returns:
+          - name: value
+            type: array[byte]
+            description: >
+                Value of the given keyword read from the given target.
+      errors:
+          - xyz.openbmc_project.Common.Error.InvalidArgument
+
+    - name: GetLocationCode
+      description: >
+          Method to get the expanded location code for the given inventory path.
+          If expanded location code is not available then unexpanded location
+          code will be returned.
+      parameters:
+          - name: inventoryPath
+            type: object_path
+            description: >
+                Inventory path of the FRU.
+      returns:
+          - name: locationCode
+            type: string
+            description: >
+                Returns the expanded location code if available, else returns
+                the unexpanded location code.
+      errors:
+          - xyz.openbmc_project.Common.Error.InvalidArgument
+
+    - name: deleteFRUVPD
+      description: >
+          Method to delete given FRU VPD and set its present property as false.
```

Comment:
Will this API delete the complete VPD?

Reply:
yes, all the persisted data of that VPD will be deleted

Reply:
Method to delete given FRU VPD persisted on cache.